File: backend/app/Modules/Lodges/Models/Lodge.php
<?php

namespace App\Modules\Lodges\Models;

use App\Modules\Secretariat\Models\Balaustre;
use App\Modules\Secretariat\Models\LodgeSession;
use App\Modules\Secretariat\Models\Member;
use App\Models\User;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class Lodge extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'name',
        'email',
        'number',
        'rite',
        'cep',
        'address_street',
        'address_number',
        'address_complement',
        'address_neighborhood',
        'city',
        'state',
    ];

    public function users(): HasMany
    {
        return $this->hasMany(User::class);
    }

    public function members(): HasMany
    {
        return $this->hasMany(Member::class);
    }

    public function lodgeSessions(): HasMany
    {
        return $this->hasMany(LodgeSession::class);
    }

    public function balaustres(): HasMany
    {
        return $this->hasMany(Balaustre::class);
    }
}

File: mobile/lib/models/lodge.dart
class Lodge {
  final int id;
  final String name;
  final String number;
  final String rite;
  final String city;
  final String state;

  Lodge({
    required this.id,
    required this.name,
    required this.number,
    required this.rite,
    required this.city,
    required this.state,
  });

  factory Lodge.fromJson(Map<String, dynamic> json) {
    return Lodge(
      id: json['id'] as int,
      name: json['name'] as String,
      number: json['number'] as String,
      rite: json['rite'] as String,
      city: json['city'] as String? ?? '',
      state: json['state'] as String? ?? '',
    );
  }
}
